core/color.cpp

```diff
@@ -435,107 +381,54 @@ Color Color::named(const String &p_name) {
 }
 
 String _to_hex(float p_val) {
-
 	int v = Math::round(p_val * 255);
 	v = CLAMP(v, 0, 255);
-	String ret;
 
+	CharType c[3] = { 0, 0, 0 };
 	for (int i = 0; i < 2; i++) {
-
-		CharType c[2] = { 0, 0 };
 		int lv = v & 0xF;
 		if (lv < 10)
-			c[0] = '0' + lv;
+			c[i] = '0' + lv;
 		else
-			c[0] = 'a' + lv - 10;
+			c[i] = 'a' + lv - 10;
 
 		v >>= 4;
-		String cs = (const CharType *)c;
-		ret = cs + ret;
 	}
 
-	return ret;
+	String cs = (const CharType *)c;
+	return cs;
 }
 
 String Color::to_html(bool p_alpha) const {
-
 	String txt;
+	if (p_alpha)
+		txt += _to_hex(a);
 	txt += _to_hex(r);
 	txt += _to_hex(g);
 	txt += _to_hex(b);
-	if (p_alpha)
-		txt = _to_hex(a) + txt;
+
```

Comment:
CSS defines HTML color codes as [RGBA](https://www.w3schools.com/cssref/func_rgba.asp), not ARGB. Godot also uses RGBA by default. Why did you swap this around?

Reply:
The functionality of this code and the old version are exactly the same. It produces a HTML color code as ARGB.
I moved the if clause up to make it more obvious as to how this string is being generated.
I don't know if it should be changed to RGBA or the effects a change like that would bring. This change was purely for readability.

Reply:
Hmm, you're right. Strange how it's `txt = x + txt` instead of `txt += x`. This should probably be changed, but someone else will also need to take a look at it.